src/org/jitsi/impl/neomedia/transform/fec/FlexFec03Mask.java

```diff
@@ -0,0 +1,262 @@
+            mask.addBit(MASK_2_K_BIT, true);
+        }
+        else if (sizeBytes > MASK_SIZE_SMALL)
+        {
+            mask.addBit(MASK_0_K_BIT, false);
+            mask.addBit(MASK_1_K_BIT, true);
+        }
+        else
+        {
+            mask.addBit(MASK_0_K_BIT, true);
+        }
+
+        return mask;
+    }
+
+    public FlexFec03BitSet getMaskWithKBits()
+    {
+        return maskWithKBits;
+    }
+
+    /**
+     * The size of the packet mask in a flexfec packet is dynamic.  This
+     * method determines the size (in bytes) of the mask in the given packet
+     * buffer and returns it
+     * @param buffer the buffer containing the mask
+     * @param maskOffset the offset in the buffer to the start of the mask
+     * @return the size of the mask, in bytes.
+     */
+    private static int getMaskSizeInBytes(byte[] buffer, int maskOffset)
+        throws MalformedMaskException
+    {
+        if ((buffer.length - maskOffset) < MASK_SIZE_SMALL)
+        {
+            throw new MalformedMaskException();
+        }
+        // The mask is always at least MASK_SIZE_SMALL bytes
+        int maskSizeBytes = MASK_SIZE_SMALL;
+        int kbit0 = (buffer[maskOffset] & 0x80) >> 7;
+        if (kbit0 == 0)
+        {
+            maskSizeBytes = MASK_SIZE_MED;
+            if ((buffer.length - maskOffset) < MASK_SIZE_MED)
+            {
+                throw new MalformedMaskException();
+            }
+            int kbit1 = (buffer[maskOffset + 2] & 0x80) >> 7;
+            if (kbit1 == 0)
+            {
+                if ((buffer.length - maskOffset) < MASK_SIZE_LARGE)
+                {
+                    throw new MalformedMaskException();
+                }
+                maskSizeBytes = MASK_SIZE_LARGE;
+            }
+        }
+        return maskSizeBytes;
+    }
+
+    /**
+     * Determine how big the mask needs to be based on the given base sequence
+     * number and the list of protected sequence numbers
+     * @param baseSeqNum the base sequence number to use for the mask
+     * @param sortedProtectedSeqNums the sequence numbers this mask should mark
+     * as protected. NOTE: this list MUST be in sorted order
+     * @return the size, in bytes, of the mask that is needed to convey
+     * the given protected sequence numbers
+     */
+    private static int getMaskSizeInBytes(int baseSeqNum, List<Integer> sortedProtectedSeqNums)
+    {
+        int largestDelta = -1;
+        for (Integer protectedSeqNum : sortedProtectedSeqNums)
+        {
+            int delta = RTPUtils.getSequenceNumberDelta(protectedSeqNum, baseSeqNum);
+            if (delta > largestDelta)
+            {
+                largestDelta = delta;
+            }
+        }
+        if (largestDelta <= 14)
+        {
+            return MASK_SIZE_SMALL;
+        }
+        else if (largestDelta <= 45)
+        {
+            return MASK_SIZE_MED;
+        }
+        return MASK_SIZE_LARGE;
```

Comment:
Do we want to signal an error if the delta is too large?

Reply:
added an exception here (like we do in the other form of this method) if the delta is too big